x-pack/plugins/apm/public/hooks/use_environments_fetcher.tsx

```diff
@@ -31,9 +35,24 @@ export function useEnvironmentsFetcher({
           },
         });
       }
+      return callApmApi('GET /internal/apm/suggestions', {
+        params: {
+          query: {
+            start,
+            end,
+            fieldName: SERVICE_ENVIRONMENT,
+            fieldValue: '',
+          },
+        },
+      });
     },
     [start, end, serviceName]
   );
 
-  return { environments: data.environments, status };
+  return {
+    environments: ((data as any)?.environments ||
+      (data as any)?.terms ||
+      []) as Environment[],
```

Comment:
This looks odd. Do you really need to cast to any here? Maybe you could match the types in the `.then` function of each `callApmApi`?
something like:
```
callApmApi('GET /internal/apm/environments',...).then(resp => resp.environments)
...
callApmApi('GET /internal/apm/suggestions',...).then(resp => resp.terms)
```

Reply:
That's a good suggestions. With the following changes, i will be able to avoid the `any` type, but would still need a cast to `Environment[]`

```
 return {
    environments: (data?.environments ?? []) as Environment[],
    status,
  };
```

I have updated the code

Reply:
@achyutjhunjhunwala in https://github.com/elastic/kibana/blob/3d0f71a35409d938c6b24ec431c3c0f635c150d2/x-pack/plugins/apm/server/routes/environments/route.ts#L30-L34

can you change to `environments: Environment[]`

Reply:
@cauemarcondes I just learned something new.

We need to be careful when we change the response of the `callApmApi` function.

If we do something like this, the API will stop appearing in the `Inspect` component of Kibana.
```
callApmApi('GET /internal/apm/environments',...).then(resp => resp.environments)
...
callApmApi('GET /internal/apm/suggestions',...).then(resp => resp.terms)
```

The `useFecher` implementation passes the `result` of the `callApmApi` to the `inspectorContext`.
Hence we must preserve the actual response

```
callApmApi('GET /internal/apm/environments',...).then(resp => { ...resp, environments: resp.environments })
...
callApmApi('GET /internal/apm/suggestions',...).then(resp => { ...resp, environments: resp.terms })
```